### PR TITLE
Revamp `RPC API For Lisk Node` page - Closes 1538

### LINKED ISSUE #1538: Update `RPC API for Lisk nodes`
### Description

- Update API for Lisk nodes RPC endpoints request and response
- Add example how to call the RPC API

### PR DESCRIPTION
**Pull request by TalhaMaliktz**
Closes #1538

### CODE REVIEW

**Review by mitsuaki-u — COMMENTED** [review]
Excellent work! Just one comment: the `chain_getGeneratorList` will now also provide a `nextAllocatedTime`(number) property once https://github.com/LiskHQ/lisk-sdk/pull/7448 is merged

**Comment by TalhaMaliktz** [comment]
> Excellent work! Just one comment: the `chain_getGeneratorList` will now also provide a `nextAllocatedTime`(number) property once [LiskHQ/lisk-sdk#7448](https://github.com/LiskHQ/lisk-sdk/pull/7448) is merged

Thanks 👍🏼  . I have updated the content based on your feedback. Also, I have added an `Events` section that needs your review.

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 1]
```diff
@@ -1,1637 +1,4113 @@
...
+:url_API_client: {url_general_endpoints}#the-api-client
 
-A reference of all actions and events that are provided by the Lisk Framework for a blockchain application.
+A reference of all RPC endpoints and events is provided by the Lisk Framework for a blockchain application.
 
-TIP: For more information about the public application endpoints, check out the xref:{url_general_endpoints}[] page.
+TIP: For more information about the public, application endpoints, and events, check out the xref:{url_general_endpoints}[] page.
 
-== Module- and plugin-specific actions and events
+== Module and plugin-specific endpoints and events
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
== Custom RPC endpoints for Modules and Plugins
```

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 2]
```diff
@@ -1,1637 +1,4113 @@
-= RPC API for Lisk nodes
-Mona Bärenfänger <mona@lightcurve.io>
+= RPC API endpoints for Lisk nodes
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
= RPC API for Lisk nodes
```

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 3]
```diff
@@ -1,1637 +1,4113 @@
...
 :url_modules: {docs_sdk}modules/index.adoc
 :url_plugins: {docs_sdk}plugins/index.adoc
+:url_subscribe_event: {url_general_endpoints}#how-to-subscribe-to-events
+:url_API_client: {url_general_endpoints}#the-api-client
 
-A reference of all actions and events that are provided by the Lisk Framework for a blockchain application.
+A reference of all RPC endpoints and events is provided by the Lisk Framework for a blockchain application.
 
-TIP: For more information about the public application endpoints, check out the xref:{url_general_endpoints}[] page.
+TIP: For more information about the public, application endpoints, and events, check out the xref:{url_general_endpoints}[] page.
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
TIP: For more information about RPC endpoints, events, and the Lisk communication architecture in general, check out the xref:{url_general_endpoints}[] page.
```

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 4]
```diff
@@ -1,1637 +1,4113 @@
...
 
-== Module- and plugin-specific actions and events
+== Module and plugin-specific endpoints and events
 
-Modules and plugins often expose additional events and actions to the application.
-All actions and events registered with the application are available as RPC endpoints.
+Modules and plugins often expose endpoints and events to the blockchain application. All endpoints and events registered with the application are available as RPC endpoints. For details about modules and plugins, see xref:{url_modules}[] and xref:{url_plugins}[].
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
Modules and plugins often expose custom endpoints to the blockchain application, which then become available as RPC endpoints.
For details about endpoints for modules and plugins, see xref:{url_modules}[] and xref:{url_plugins}[].
```

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 5]
```diff
@@ -1,1637 +1,4113 @@
...
 
-The actions and events of the official modules and plugins are described on their respective reference pages, see xref:{url_modules}[] and xref:{url_plugins}[].
+== Usage
+Endpoints and events exposed by Lisk are categorized based on various namespaces. These namespaces point toward various, modules, plugins, and default application-specific endpoints and events. To invoke an endpoint or to subscribe to an event, combine the name of an endpoint/event with its namespace. For example:
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
RPC endpoints and events exposed by a blockchain application are categorized based on various namespaces. These namespaces point toward various, modules, plugins, and default application-specific domains.
To invoke an endpoint or to subscribe to an event, combine the name of an endpoint/event with its namespace.
For example:
```

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 6]
```diff
@@ -1,1637 +1,4113 @@
...
 
-Actions are invoked to receive specific data from the blockchain application.
-Actions are part of the public request / response API, and are invoked via RPCs.
+== Endpoints
 
-=== `app:getRegisteredActions`
+Endpoints are invoked to fetch specific data for an application. Lisk node endpoints are public and can be invoked via RPCs. Lisk supports three modes of communication for invoking an endpoint as discussed in xref:{url_API_client}[The API client] section. The default RPC endpoints have been categorized based on various namespaces, as follows:
 
-Returns a list of all actions known to the application.
+. Chain
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
. <<chain>>
```

adding internal anchors here might be beneficial

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 7]
```diff
@@ -1,1637 +1,4113 @@
...
+
+
+
+
+
+
 == Events
 
-Events are part of the public publish / subscribe API of a blockchain application.
+Events exposed by the Lisk RPC node can be subscribed as explained in the xref:{url_subscribe_event}[Subscribing to the events] section. To subscribe to an event, a WebSocket or Inter-Process Communication protocol must be used. The default events exposed by the Lisk Framework are described below:
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Are these all events that are available? E.g. I think it was mentioned that every transactions triggers an event, announcing is the tx failed or was successful, is this included here?

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 8]
```diff
@@ -1734,9 +4270,1722 @@ Emitted when a block is deleted from the blockchain.
...
-    [ "0a14e2950a9f07b44e724df2129360cc140293c08308120208001a020800220208002a3a0a190a0a67656e657369735f35371800200228003080a094a58d1d121d0a14e2950a9f07b44e724df2129360cc140293c083081080a094a58d1dc23e020a00" ]
+   "transaction": {
+   "id": "95b18ca901c910ea34d5df8896f6a9bc477f773ba5d0ff08c500711c15efb1db",
+   ...
+   }
 }
 ----
-====
\ No newline at end of file
+====
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc:4282]: Please delete the old content, instead of commenting out.

**Review by Tschakki — CHANGES_REQUESTED** [review]
Looking good, just a few small suggestions :)

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 8 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc:4282]: Will do.

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 7 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: In total there are eight events as mentioned in the [events.ts file](https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/engine/events.ts). Due to the unavailability of some content, I have left a note [here](https://github.com/LiskHQ/lisk-docs/issues/1538) after discussing it with Shu

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 1 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Agreed, text updated.

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 3 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Agreed, text updated

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 4 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Agreed, text updated

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 6 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Agreedl text updated

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 9]
```diff
@@ -1,1637 +1,4118 @@
...
+
+
+
+
+
+
 == Events
 
-Events are part of the public publish / subscribe API of a blockchain application.
+Events exposed by the Lisk RPC node can be subscribed as explained in the xref:{url_subscribe_event}[subscribing to the events] section. To subscribe to an event, a WebSocket or Inter-Process Communication protocol must be used. The default events exposed by the Lisk Framework are described below:
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc:4055]: A list of the events namespaces, analog to the list for the endpoints at line 42-47 would be nice

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 10]
```diff
@@ -1,1637 +1,4118 @@
...
-== Module- and plugin-specific actions and events
+== Custom RPC endpoints for Modules and Plugins
 
-Modules and plugins often expose additional events and actions to the application.
-All actions and events registered with the application are available as RPC endpoints.
+Modules and plugins often expose custom endpoints to the blockchain application, which then become available as RPC endpoints.
+For details about endpoints for modules and plugins, see xref:{url_modules}[] and xref:{url_plugins}[].
 
-The actions and events of the official modules and plugins are described on their respective reference pages, see xref:{url_modules}[] and xref:{url_plugins}[].
+== Usage
```
Comment (Tschakki) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Aliases and usage examples are also described on the SDK RPC page updated in #1513 , please add links to the relevant sections at this place. Alternatively, you can update this section also in PR #1513, as you prefer :)

**Review by Tschakki — APPROVED** [review]
Thank you! LGTM :)

Just added a few last suggestions

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 9 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc:4055]: Done

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc (hunk 10 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Ummmm, various Links to the **API Client** and **Subscribing Event** sections are present in this PR. Also, I already updated the Alias section in PR #1513 so not sure about the suggested change there.  Should I add a tip about the API Client and Subscribing Events section? IMHO that would be redundant. What do you think?

Comment (Tschakki) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Hmm, yeah I think the content under the heading `Usage` is maybe not really fitting here, and you are right it is a little redundant to the `aliases` section in #1513. So maybe we can remove the content here, or merge it with the `aliases` section.

On the other hand, the content in the next section `Endpoints` in line 40 explains the usage of the RPC API, and also provides a link to the usage example in #1513. I think we could move this up under the `usage` section, if we want to keep this section(except of the last sentence in line 40).

- Btw we normally use a new line for every new sentence, it makes overview and review easier :)

Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]: Thanks, I have removed the `Usage` section as you suggested. I have also incorporated the new line convention in this file. Will follow it in the future as well :)

**Pull request merged**